extensions/amp-story-interactive/0.1/test/test-amp-story-interactive-img-poll.js

```diff
@@ -134,5 +147,83 @@ describes.realWin(
         }).to.throw(/Improper number of options/);
       });
     });
+
+    it('should handle the percentage pipeline', async () => {
+      env.sandbox
+        .stub(requestService, 'executeRequest')
+        .resolves(getMockInteractiveData());
+
+      ampStoryPoll.element.setAttribute('endpoint', 'http://localhost:8000');
+
+      addConfigToInteractive(ampStoryPoll, 2);
+      await ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+
+      expect(ampStoryPoll.getOptionElements()[0].innerText).to.contain('50%');
+      expect(ampStoryPoll.getOptionElements()[1].innerText).to.contain('50%');
+    });
+
+    it('should handle the percentage pipeline with scrambled data', async () => {
+      const NUM_OPTIONS = 4;
+      env.sandbox
+        .stub(requestService, 'executeRequest')
+        .resolves(getMockScrambledData());
+
+      ampStoryPoll.element.setAttribute('endpoint', 'http://localhost:8000');
+
+      addConfigToInteractive(ampStoryPoll, NUM_OPTIONS);
+      await ampStoryPoll.buildCallback();
+      await ampStoryPoll.layoutCallback();
+
+      const expectedPercentages = [10, 20, 30, 40];
+      for (let i = 0; i < NUM_OPTIONS; i++) {
+        const expectedText = `${expectedPercentages[i]}%`;
+        expect(ampStoryPoll.getOptionElements()[i].innerText).to.contain(
+          expectedText
+        );
+      }
```

Comment:
Nit: It's usually a good idea to not have any logic in the tests (eg: for loops) so that you can see what line fails the test immediately.

Reply:
Can handle in separate PR that addresses not only image components